webapp/src/test/java/rocks/metaldetector/security/XSSFilterTest.java

```diff
@@ -0,0 +1,47 @@
+package rocks.metaldetector.security;
+
+import org.assertj.core.api.WithAssertions;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.MockedStatic;
+import org.mockito.Mockito;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.mock.web.MockFilterChain;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletResponse;
+import java.io.IOException;
+
+import static org.mockito.ArgumentMatchers.any;
+
+@ExtendWith(MockitoExtension.class)
+class XSSFilterTest implements WithAssertions {
+
+  private final XSSFilter underTest = new XSSFilter();
+
+  private final MockHttpServletRequest request = new MockHttpServletRequest();
+  private final ServletResponse response = new MockHttpServletResponse();
+  private final FilterChain filterChain = new MockFilterChain();
+
+  @Test
+  @DisplayName("XSSUtils are called to remove bad JS from request body")
```

Comment:
The JavaScript part is confusing, because there is no JavaScript stuff in the request body.

```suggestion
  @DisplayName("XSSUtils are called to sanitize request body")
```